packages/swing-store/src/snapStore.js

```diff
@@ -151,11 +191,35 @@ export function makeSnapStore(
       template: 'atomic-XXXXXX',
     });
     try {
+      // This nested await is safe because "synchronous-throw-impossible"
+      // This relies on `writeContents` itself being a
+      // "synchronous-throw-impossible" function, which we modified it to be.
+      // eslint-disable-next-line @jessie.js/no-nested-await
       await writeContents(tmpFilePath);
       const target = resolve(root, baseName);
-      await rename(tmpFilePath, target);
+      // This nested await is safe because "IIAFE" (immediately invoked
+      // async function expression). An IIAFE cannot synchronously throw
+      // and is therefore a special case of "synchronous-throw-impossible".
+      //
+      // The original was
+      // ```js
+      // await rename(tmpFilePath, target);
+      // ```
+      // which we could not locally verify to be safe. `rename` is a parameter
+      // of an exported function, which we should therefore be defensive
+      // against. If it throws, `unlink(tmp)` could be reached synchronously,
+      // whereas otherwise it would be reached asynchronously. Because
+      // `unlink` is also a parameter of an exported function, we cannot
+      // locally verify it to be stateless. In fact, from the name, we
+      // expect it to be stateful. By changing the awaited expression to an
+      // IIAFE, we ensure that the `unlink` can only be reached asynchronously.
+      //
+      // eslint-disable-next-line @jessie.js/no-nested-await
+      await (async () => rename(tmpFilePath, target))();
```

Comment:
I'm not sure what to say about this file. It may be that keeping all this commentary inline is cost-effective.